scheduler/reconcile.go

```diff
@@ -201,29 +212,28 @@ func (a *allocReconciler) Compute() *reconcileResults {
 	}
 
 	// Detect if the deployment is paused
-	if a.deployment != nil {
-		a.deploymentPaused = a.deployment.Status == structs.DeploymentStatusPaused ||
-			a.deployment.Status == structs.DeploymentStatusPending
-		a.deploymentFailed = a.deployment.Status == structs.DeploymentStatusFailed
-	}
-	if a.deployment == nil {
-		// When we create the deployment later, it will be in a pending
-		// state. But we also need to tell Compute we're paused, otherwise we
-		// make placements on the paused deployment.
-		if a.job.IsMultiregion() && !(a.job.IsPeriodic() || a.job.IsParameterized()) {
-			a.deploymentPaused = true
-		}
-	}
+	a.computePaused()
+
+	// Reconcile each group and determine if the all groups have deployed completely.
+	deploymentComplete := a.computeDeploymentComplete(m)
+
+	a.computeDeploymentUpdates(deploymentComplete)
 
-	// Reconcile each group
-	complete := true
+	return a.result
+}
+
+func (a *allocReconciler) computeDeploymentComplete(m allocMatrix) bool {
+	deploymentComplete := true
```

Comment:
#Refactoring review

Reply:
If you've moved this into a method named `computeDeploymentComplete`, you probably don't need to also change the name of this variable to something so long.